resources/scponly_user.rb

```diff
@@ -0,0 +1,99 @@
+#true To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponly_user
+
+default_action :create
+
+property :write_dir, String, default: 'incoming'
+property :public_key, String
+property :private_key, String
+property :chroot, [true, false], default: true
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  if new_resource.chroot
+    user new_resource.name do
+      gid 'scponly'
+      manage_home true
+      shell '/usr/sbin/scponlyc'
+    end
+
+    directory "/home/#{new_resource.name}//#{new_resource.write_dir}"do 
+      owner 'root'
+      group 'scponly'
+    end
+
+    altroot = "/home/#{new_resource.name}"
+    binaries = %w(/bin/chgrp /bin/chmod /bin/chown /bin/ln /bin/ls /bin/mkdir /bin/mv /bin/rm /bin/rmdir /usr/bin/scp /usr/sbin/scponlyc)
+
+    cookbook_file '/tmp/chroot.sh' do
+      source 'chroot.sh'
+      mode 0755
+    end
+
+    execute 'Build chroot jail' do
+      command "bash /tmp/chroot.sh #{altroot} #{binaries.join(' ')}"
+    end
+
+    directory "#{altroot}/etc"
+    %w(ld.so.cache ld.so.conf group).each do |c|
+      bash "Chroot /etc/#{c}" do
+        code "cp /etc/#{c} #{altroot}/etc/"
+      end
+    end
+
+    bash 'Chroot /etc/passwd' do
+      code "cat /etc/passwd | grep #{new_resource.name} > #{altroot}/etc/passwd"
+    end
```

Comment:
What are you trying to do here? It seems as though we could do this differently.

Reply:
Per the BUILDING-JAILS.txt, the chrooted environment needs a slightly modified passwd entry for the chrooted user and I figured this was the easiest way to grab that line.

Reply:
Do you only need a single line entry for the ``passwd`` file in the chroot environment?

Reply:
As I understand it yes but the chrooted `passwd` is a subset of the system file

Reply:
Here's what I'd do and seems to work instead:
```ruby
execute "grep #{new_resource.name} /etc/passwd > #{altroot}/etc/passwd" do
  creates "#{altroot}/etc/passwd"
end
```